editor/editor_node.cpp

```diff
@@ -4940,24 +4984,33 @@ void EditorNode::_load_docks_from_config(Ref<ConfigFile> p_layout, const String
 				atidx = k;
 				break;
 			}
+
 			if (atidx == -1) { // Well, it's not anywhere.
 				continue;
 			}
 
 			if (atidx == i) {
 				dock_slot[i]->move_child(node, 0);
-				continue;
-			}
+			} else {
+				dock_slot[atidx]->remove_child(node);
 
-			dock_slot[atidx]->remove_child(node);
+				if (dock_slot[atidx]->get_tab_count() == 0) {
+					dock_slot[atidx]->hide();
+				}
+				dock_slot[i]->add_child(node);
+				dock_slot[i]->move_child(node, 0);
+				dock_slot[i]->set_tab_title(0, TTRGET(node->get_name()));
+				dock_slot[i]->show();
+			}
 
-			if (dock_slot[atidx]->get_tab_count() == 0) {
-				dock_slot[atidx]->hide();
+			WindowWrapper *wrapper = Object::cast_to<WindowWrapper>(node);
```

Comment:
This breaks loading the floating docks. `node` is not a WindowWrapper, it's the dock itself.